src/screens/ChannelListScreenV2/index.tsx

```diff
@@ -8,79 +9,74 @@ import ChannelListScreen from '../ChannelListScreen';
 
   const navigateToContactScreen = () => {
     navigation.navigate('ContactScreen');
   };
 
-  const onTabSelected = (index: number) => {
-    setSelectedTab(index);
-    if (index === 0) {
-      refreshSignedChannelList();
-    } else if (index === 1) {
-      refreshAnonymousChannelList();
-    }
-  };
+  React.useEffect(() => {
+    if (isFocused) refresh('channelList');
+  }, [isFocused]);
 
   return (
     <>
       <StatusBar translucent={false} />
-      <ScrollView contentContainerStyle={{flexGrow: 1}}>
-        <View style={{height: 52}}>
-          <Search animatedValue={0} onPress={navigateToContactScreen} />
-        </View>
-        <HorizontalTab
-          selectedTab={selectedTab}
-          onSelectedTabChange={onTabSelected}
-          tabs={[
-            {
-              key: 0,
-              tabElement: (
+      <FlatList
+        data={[]}
+        keyExtractor={(_e, i) => `dom${i.toString()}`}
+        ListEmptyComponent={null}
+        renderItem={null}
+        contentContainerStyle={{flexGrow: 1}}
+        ListHeaderComponent={
+          <>
+            <View style={{height: 52}}>
+              <Search animatedValue={0} onPress={navigateToContactScreen} />
+            </View>
+
+            <HorizontalTab
+              selectedTab={selectedTab}
+              onSelectedTabChange={setSelectedTab}
+              tabs={[
                 <ChannelListTabItem
                   key={0}
                   name={`@${profile?.username}`}
                   picture={profile?.profile_pic_path}
                   unreadCount={signedChannelUnreadCount}
-                  testID="signed-channel-list-tab-item"
-                />
-              )
-            },
-            {
-              key: 1,
-              tabElement: (
+                  type="SIGNED"
+                  testID="horizontal-tab-0"
+                />,
                 <ChannelListTabItem
                   key={1}
                   name="Anonymous"
                   picture={AnonymousProfile}
                   unreadCount={anonymousChannelUnreadCount}
-                  testID="anonymous-channel-list-tab-item"
+                  type="ANONYMOUS"
+                  testID="horizontal-tab-1"
                 />
-              )
-            }
-          ]}
-        />
-        <View style={{display: selectedTab === 0 ? 'flex' : 'none'}}>
-          <ChannelListScreen />
-        </View>
-        <View style={{display: selectedTab === 1 ? 'flex' : 'none'}}>
-          <AnonymousChannelListScreen />
-        </View>
-      </ScrollView>
+              ]}
+            />
+
+            <View style={{display: selectedTab === 0 ? 'flex' : 'none'}}>
+              <ChannelListScreen />
+            </View>
+            <View style={{display: selectedTab === 1 ? 'flex' : 'none'}}>
+              <AnonymousChannelListScreen />
+            </View>
+          </>
+        }
+      />
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The switch from `ScrollView` to `FlatList` is a good performance improvement for long lists. However, there's an issue with the `FlatList` usage. The `data` prop is set to an empty array and `renderItem` is set to `null`, which means this `FlatList` won't render any items. If you're using `FlatList` just for the `ListHeaderComponent`, consider switching back to `ScrollView` or another layout component that better suits your needs. 

Also, the conditional rendering of `<ChannelListScreen />` and `<AnonymousChannelListScreen />` based on `selectedTab` could lead to unnecessary unmounting and remounting of these components when switching tabs. This could potentially cause performance issues if these components have expensive setup or teardown operations. Consider using a visibility style (e.g., `opacity: 0` and `height: 0`) instead of `display: 'none'` to hide inactive tabs while keeping their state and DOM presence.

```diff
- <View style={{display: selectedTab === 0 ? 'flex' : 'none'}}>
+ <View style={{opacity: selectedTab === 0 ? 1 : 0, height: selectedTab === 0 ? 'auto' : 0}}>
    <ChannelListScreen />
- </View>
- <View style={{display: selectedTab === 1 ? 'flex' : 'none'}}>
+ </View>
+ <View style={{opacity: selectedTab === 1 ? 1 : 0, height: selectedTab === 1 ? 'auto' : 0}}>
    <AnonymousChannelListScreen />
- </View>
+ </View>
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->